rust/processor/src/processors/econia_processor.rs

```diff
@@ -622,14 +656,68 @@ impl ProcessorTrait for EconiaTransactionProcessor {
                     )?);
                 }
             }
+            // Index transaction write set.
+            let info = &txn.info.as_ref().expect("No transaction info");
+            for change in &info.changes {
+                match change.change.as_ref().expect("No transaction changes") {
+                    Change::WriteResource(resource) => {
+                        if resource.r#type.as_ref().expect("No resource type")
+                            == &market_accounts_type
+                        {
+                            let data: serde_json::Value = serde_json::from_str(&resource.data)
+                                .expect("Failed to parse MarketAccounts");
+                            let map_field = data.get("map").expect("No map field");
+                            market_account_handles.push(MarketAccountHandle {
+                                user: resource.address.clone(),
+                                handle: opt_value_to_string(map_field.get("handle"))?,
+                                creation_time: time,
+                            })
+                        }
+                    },
+                    Change::WriteTableItem(write) => {
+                        let table_data = write.data.as_ref().expect("No WriteTableItem data");
+                        if table_data.value_type
+                            != format!("{}::user::MarketAccount", econia_address)
+                        {
+                            continue;
+                        }
+                        let table_key: serde_json::Value = serde_json::from_str(&table_data.key)
+                            .expect("Failed to parse market account ID to JSON");
+                        let market_account_id = u128::from_str(
+                            &table_key
+                                .as_str()
+                                .expect("Failed to parse market account ID to string"),
+                        )
+                        .expect("Failed to parse market account ID to u128");
+                        let data: serde_json::Value = serde_json::from_str(&table_data.value)
+                            .expect("Failed to parse MarketAccount");
+                        balance_updates.push(BalanceUpdate {
+                            txn_version: txn_version.into(),
+                            handle: write.handle.to_string(),
```

Comment:
We should include the user id associated with the handle in this table, so that events notifying balance updates contain it--the handle is kind of useless without referring to the other table. Since the ID associated with the handle we can save the look-up and make things easier for people by including it in this table too.

Reply:
This is accounted for via the `balance_updates` natural join view (see the uo.sql migration).

`market_account_handles`:

<img width="681" alt="Screenshot 2023-10-02 at 15 41 49" src="https://github.com/econia-labs/aptos-indexer-processors/assets/43892045/e950fdc0-bcc5-476d-bbab-44f4f30d9e62">

`balance_updates_by_handle`:

<img width="686" alt="Screenshot 2023-10-02 at 15 44 25" src="https://github.com/econia-labs/aptos-indexer-processors/assets/43892045/3dc9a682-12a4-4ac7-bf7b-48e7754ccd20">

`balance_updates` (the natural join view function):

<img width="681" alt="Screenshot 2023-10-02 at 15 42 01" src="https://github.com/econia-labs/aptos-indexer-processors/assets/43892045/4cfe5abf-50d4-40c9-a9b3-f590ca810858">

Reply:
We can't attach notifications to that view though, and cross-referencing every event to that view in order to get the user address is both expensive (it's a full table join) and unnecessary (the data can be included).

Are you expecting that people will perform a full table join every time they get a balance update event in order to get the user's address?

Reply:
There are multiple possible solutions here, none of which require that the processor cache extra data or read from the database:

1. Create a trigger that simply sends out `balance_update_by_handle` entries verbatim. In this case, the listener simply has to look up their handle in `market_account_handles` one time and then knows which events to retain in the future
2. Create a trigger that sends out the equivalent of an entry in `balance_updates`, but without doing a join. The function needs to take a `balance_updates_by_handle` entry and lookup the handle in `market_account_handles`, then substitute in the corresponding user field. This can be done purely in migrations, though it will require that the db insertion does market account handle insertions *before* balance updates (currently they happen after per the `build_transaction()`.

I lean toward (1) as it eliminates internal dependencies and is a simple trigger

Reply:
If we go with option 1 then I think we should just get rid of the natural join view. I don't think it has such value or utility that it should be part of the API surface area.